python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,498 @@
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results as a struct with attributes:
+
+            ``exposure``
+                Calibrated exposure, with pixels in nJy units.
+                (`lsst.afw.image.Exposure`)
+            ``source``
+                Sources that were measured on the exposure calibrated fluxes
+                and magnitudes.
+                (`astropy.table.Table`)
+            ``source_footprints``
+                Footprints of sources that were measured on the exposure.
+                (`lsst.afw.table.SourceCatalog`)
+        """
+        if result is None:
+            result = pipeBase.Struct()
+        if id_generator is None:
+            id_generator = lsst.meas.base.IdGenerator()
+
+        result.exposure = self.snap_combine.run(exposures).exposure
+
+        # Calibrate the image, so it's on the same units as the background.
+        result.exposure.maskedImage = photo_calib.calibrateImage(result.exposure.maskedImage)
+        result.exposure.maskedImage -= background.getImage()
+        # Uncalibrate so that we do the measurements in instFlux, because
+        # we don't have a way to identify measurements as being in nJy (TODO!).
+        result.exposure.maskedImage /= photo_calib.getCalibrationMean()
+
+        result.exposure.setPsf(psf)
+        result.exposure.setApCorrMap(ap_corr)
+        result.exposure.setWcs(wcs)
+        result.exposure.info.setSummaryStats(summary_stats)
+        # Note: we don't set photoCalib here, because we have to use it below
+        # to calibrate the image and catalog, and thus the image will have a
+        # photoCalib of exactly 1.
+
+        result.source_footprints = self._find_sources(result.exposure, background, id_generator)
+        result.source_footprints["visit"] = result.exposure.visitInfo.id
+        result.source_footprints["detector"] = result.exposure.info.getDetector().getId()
+        result.background = background
+        self._apply_photo_calib(result.exposure, result.source_footprints, photo_calib)
+
+        # TODO: do we need this, or can we drop LocalWcs/LocalPhotoCalib?
+        self.post_calculations.run(result.source_footprints, result.exposure)
+        result.source = result.source_footprints.asAstropy()
+
+        return result
+
+    def _find_sources(self, exposure, background, id_generator):
+        """Detect and measure sources on the exposure.
+
+        Parameters
+        ----------
+        exposures : `lsst.afw.image.Exposure`
+            Exposure to detect and measure sources on; must have a valid PSF.
+        background : `lsst.afw.math.BackgroundList`
+            Background that was fit to the exposure during detection;
+            modified in-place during subsequent detection.
+        id_generator : `lsst.meas.base.IdGenerator`
+            Object that generates source IDs and provides random seeds.
+
+        Returns
+        -------
+        source
```

Comment:
I don't like `source` as the name of the catalog, it seems like a single source.

Reply:
I was following along from the name in [DMTN-172](https://dmtn-172.lsst.io), but maybe that doesn't make sense in code? I wonder if we shouldn't change that DMTN recommendation to be `sources`, since it does suggest `stars` there?